ext/drreg/drreg.c

```diff
@@ -385,14 +385,20 @@ drreg_event_bb_analysis(void *drcontext, void *tag, instrlist_t *bb, bool for_tr
             /* DRi#1849: COND_SRCS here includes addressing regs in dsts */
             if (instr_reads_from_reg(inst, reg, DR_QUERY_INCLUDE_COND_SRCS))
                 value = REG_LIVE;
-            /* make sure we don't consider writes to sub-regs */
-            else if (instr_writes_to_exact_reg(inst, reg, DR_QUERY_INCLUDE_COND_SRCS)
-                     /* A write to a 32-bit reg zeroes the top 32 bits for x86_64 and
-                      * aarch64.
-                      */
-                     IF_X64(||
-                            instr_writes_to_exact_reg(inst, reg_64_to_32(reg),
-                                                      DR_QUERY_INCLUDE_COND_SRCS)))
+            /* Make sure we don't consider writes to sub-regs. i#6417: in case
```

Comment:
Please make a separate comment for this.  Having it on the end of the line about sub-regs makes it seem like it's related to sub-regs.